### PR TITLE
physical_resources_check:fix serial generator

### PR DESCRIPTION
**Pull request by qingwangrh**
Disk serial should exclude special characters.

ID:2144439
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=physical_resources_check --iothread_scheme=roundrobin --nr_iothreads=2 --platform=x86_64 --guestname=RHEL.9.2.0 --driveformat=virtio_blk --nicmodel=virtio_net --imageformat=qcow2 --machines=q35 --customsparams="vm_mem_limit = 12G\nimage_aio=threads" --firmware=default_bios --netdst=virbr0 --debug --clone=no --nrepeat=5
 (1/5) repeat1.Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.physical_resources_check.q35: STARTED
 (1/5) repeat1.Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.physical_resources_check.q35: PASS (37.51 s)
 (2/5) repeat2.Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.physical_resources_check.q35: STARTED
 (2/5) repeat2.Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.physical_resources_check.q35: PASS (39.10 s)
 (3/5) repeat3.Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.physical_resources_check.q35: STARTED
 (3/5) repeat3.Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.physical_resources_check.q35: PASS (39.11 s)
 (4/5) repeat4.Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.physical_resources_check.q35: STARTED
 (4/5) repeat4.Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.physical_resources_check.q35: PASS (39.04 s)
 (5/5) repeat5.Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.physical_resources_check.q35: STARTED
 (5/5) repeat5.Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.physical_resources_check.q35: PASS (38.59 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by qingwangrh** [comment]
@YongxueHong @XueqiangWei @zhenyzha Please help to review,thanks.

File: qemu/tests/physical_resources_check.py [hunk 1]
```diff
@@ -207,12 +208,8 @@ def verify_machine_type():
     if params.get("catch_serial_cmd") is not None:
         length = int(params.get("length", "20"))
         id_leng = random.randint(0, length)
-        drive_serial = ""
-        convert_str = "!\"#$%&\'()*+./:;<=>?@[\\]^`{|}~"
-        drive_serial = utils_misc.generate_random_string(id_leng,
-                                                         ignore_str=",",
-                                                         convert_str=convert_str)
-
+        ignore_str = string.punctuation.replace("-", "").replace("_", "")
```
Comment (XueqiangWei) [inline qemu/tests/physical_resources_check.py:211]: `import string

result = string.punctuation
print(result)
!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~`

Comment (XueqiangWei) [reply qemu/tests/physical_resources_check.py:211]: OK, I see. Agree with you, select drive_serial characters from ascii_letters + digits + "-"  + "_".

**Comment by zhenyzha** [comment]
Interesting cases actually it's only x86.
But I did some tests and found that it has loose logic conditions. Skip if some cmd has no return value.
So I believe it can also pass on to other platforms.
But because it is some basic check, there is no platform difference. So I think the current code is ok for me. There is no need to enable this test on ARM.

LGTM.Acked-by: Zhenyu Zhang [zhenyzha@redhat.com](mailto:zhenyzha@redhat.com)

**Review by zhenyzha — APPROVED** [review]

**Review by XueqiangWei — APPROVED** [review]

**Comment by XueqiangWei** [comment]
@YongxueHong Please help take a look at it. Many  thanks.

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Pull request merged**